commands/registrySettings.ts

```diff
@@ -1,36 +1,40 @@
 import * as vscode from 'vscode';
 import { AzureRegistryNode } from "../explorer/models/azureRegistryNodes";
 import { CustomRegistryNode } from "../explorer/models/customRegistryNodes";
 import { DockerHubOrgNode } from "../explorer/models/dockerHubNodes";
 
+const defaultRegistryPathKey = "defaultRegistryPath";
+const defaultRegistryKey = "defaultRegistry";
+const combineRegistryPathsKey = "combineRegistryPathsFlag"
+
 export async function setRegistryAsDefault(node: CustomRegistryNode | AzureRegistryNode | DockerHubOrgNode): Promise<void> {
     let registryName: string;
     if (node instanceof DockerHubOrgNode) {
         registryName = node.namespace;
     } else if (node instanceof AzureRegistryNode) {
         registryName = node.registry.loginServer || node.label;
-    } else { //CustomREgistryNode
+    } else if (node instanceof CustomRegistryNode) {
         registryName = node.registryName;
+    } else {
+        assert.equal(false, true, "Type of registry node encountered not handled");
     }
 
     const configOptions: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration('docker');
-    await configOptions.update('defaultRegistryPath', registryName, vscode.ConfigurationTarget.Workspace);
-    vscode.window.showInformationMessage(`Update defaultRegistryPath to ${registryName}`);
+    await configOptions.update(defaultRegistryPathKey, registryName, vscode.ConfigurationTarget.Workspace);
+    vscode.window.showInformationMessage(`Updated defaultRegistryPath to ${registryName}`);
 }
 
 export async function consolidateDefaultRegistrySettings(): Promise<void> {
     const configOptions: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration('docker');
-    let defaultRegistryPath = configOptions.get('defaultRegistryPath', '');
-    let defaultRegistry = configOptions.get('defaultRegistry', '');
+    const combineRegistryPaths: boolean = configOptions.get(combineRegistryPathsKey);
+    let defaultRegistryPath: string = configOptions.get(defaultRegistryPathKey, '');
+    let defaultRegistry: string = configOptions.get(defaultRegistryKey, '');
 
-    if (defaultRegistry) {
-        if (defaultRegistryPath) { // combine both the settings, then notify the user
-            await configOptions.update('defaultRegistryPath', `${defaultRegistry}/${defaultRegistryPath}`, vscode.ConfigurationTarget.Workspace);
-        }
-        if (!defaultRegistryPath) {// assign defaultRegistry to defaultRegistryPath
-            await configOptions.update('defaultRegistryPath', `${defaultRegistry}`, vscode.ConfigurationTarget.Workspace);
-        }
-        await configOptions.update('defaultRegistry', undefined, vscode.ConfigurationTarget.Workspace);
+    if (defaultRegistry && combineRegistryPaths) {
+        let updatedPath = defaultRegistryPath ? `${defaultRegistry}/${defaultRegistryPath}` : `${defaultRegistry}`;
+        await configOptions.update('defaultRegistryPath', updatedPath, vscode.ConfigurationTarget.Global);
```

Comment:
Not using the constants.  BTW, my own guidelines for when to use a constant are essentially: 1) if used multiple times, 2) if a constant would help document it.